test/subspace-test-service/src/mock.rs

```diff
@@ -142,6 +142,10 @@ impl MockPrimaryNode {
             _,
         >::new(Box::new(fraud_proof_block_import), client.clone());
 
+        // The `maintain-bundles-stored-in-last-k` worker here is different from the one in the production code
+        // that it subscribes the `block_importing_notification_stream`, which is intended to ensure the bundle
+        // validator's `recent_stored_bundles` info must be updated when a new primary block is produced, this
```

Comment:
If I understand correctly, this means the update of `recent_stored_bundles` for the incoming primary block must be complete before producing the next primary block.

Reply:
That is right.